File: apps/sdkwork-appstore-flutter-mobile/packages/sdkwork_appstore_flutter_mobile_updates/lib/src/services/updates_service.dart
import 'package:sdkwork_appstore_flutter_mobile_core/sdkwork_appstore_flutter_mobile_core.dart';

import '../models/updates_models.dart';

/// Updates service.
///
/// The App Store app SDK clients are injected by the root bootstrap
/// (`APP_SDK_INTEGRATION_SPEC.md`); this package never constructs a client and
/// never issues raw HTTP.
class UpdatesService {
  const UpdatesService({required this.clients});

  final AppstoreAppSdkClients clients;

  String get capability => 'updates';

  UpdatesPageResult<Never> empty() =>
      const UpdatesPageResult<Never>(items: <Never>[]);
}

File: apps/sdkwork-appstore-mini-program/packages/sdkwork-appstore-mp-updates/src/services/updatesService.ts
import type { AppstoreAppClient } from "@sdkwork/appstore-mp-core";

import type { UpdatesPageResult } from "../types/updatesModels.js";

/**
 * Updates service.
 *
 * The generated App Store app SDK client is injected by the root bootstrap
 * (`APP_SDK_INTEGRATION_SPEC.md`); this package never constructs a client and
 * never issues raw HTTP.
 */
export interface UpdatesService {
  readonly capability: "updates";
}

export function createUpdatesService(client: AppstoreAppClient): UpdatesService {
  void client;
  return { capability: "updates" };
}

export function emptyUpdatesResult(): UpdatesPageResult<never> {
  return { items: [] };
}
